# Installation of software in a custom directory

In:
```
cd /fs/scratch/PAS2658/Alexis/Annotation/Software
```

Installing **BBTools**: a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data https://jgi.doe.gov/data-and-tools/bbtools/
```
wget https://sourceforge.net/projects/bbmap/files/BBMap_39.06.tar.gz
tar -xvzf BBMap_39.06.tar.gz
rm BBMap_39.06.tar.gz
# testing BBTools:
bash ./bbmap/bbmerge.sh
```

Installing **FastQC v0.11.9** https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
```
apptainer pull fastqc.sif docker://quay.io/biocontainers/fastqc:0.12.1--hdfd78af_0
# Does FastQC work?
apptainer exec fastqc.sif fastqc --help
```

Installing **MultiQC v1.13** https://multiqc.info
```
apptainer pull /fs/scratch/PAS2658/Alexis/Annotation/Software/MultiQC.sif docker://ewels/multiqc
# Does MultiQC work?
apptainer exec MultiQC.sif multiqc --help
```

Installing **Funannotate 1.8.17**: Fuannotate: pipeline for annotating fungal genomes https://github.com/nextgenusfs/funannotate
```
apptainer pull Funnanotate.sif docker://nextgenusfs/funannotate
# Testing Funannotate
apptainer exec Funnanotate.sif funannotate #check spelling
```

Installing **GeneMark-ES/ET**: (you will need to generate your own license, Jonathan will show you how to do that, do not install before class) Alexandre Lomsadze et al Gene identification in novel eukaryotic genomes by self-training algorithm http://nar.oxfordjournals.org/content/33/20/6494.abstract  http://topaz.gatech.edu/GeneMark/license_download.cgi
#GeneMark-ES/ET/EP ver 4.70_lic

go to link, enter information for:
Software:
GeneMark-ES/ET/EP+ ver 4.72_lic
Operating system:
LINUX 64 kernel 3.10 - 5

enter institutional info and accept

Download here -> right click and copy link. then enter into terminal wget [link] - line 56

then the key use 64 bit and right click copy link, enter into terminal - line 62


# LINUX 64 kernel 3.10 - 5
```
wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_qJroM/gmes_linux_64_4.tar.gz

tar -xvzf gmes_linux_64_4.tar.gz
rm gmes_linux_64_4.tar.gz
cd gmes_linux_64_4

wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_qJroM/gm_key_64.gz
gunzip gm_key_64.gz
cp gm_key_64 ~/.gm_key

# checking if GeneMark works
./predict_genes.pl
```

Installing **Agat v1.3.0**: Another Gtf/Gff Analysis Toolkit https://github.com/NBISweden/AGAT
```
cd ..
apptainer pull AGAT.sif docker://quay.io/biocontainers/agat:1.3.0--pl5321hdfd78af_0
# Checking if AGAT works
apptainer exec AGAT.sif agat_convert_sp_gxf2gxf.pl --help
```

Installing **BUSCO v5.6.1**: Mosè Manni, Matthew R Berkeley, Mathieu Seppey, Felipe A Simão, Evgeny M Zdobnov, BUSCO Update: Novel and Streamlined Workflows along with Broader and Deeper Phylogenetic Coverage for Scoring of Eukaryotic, Prokaryotic, and Viral Genomes. Molecular Biology and Evolution, Volume 38, Issue 10, October 2021, Pages 4647–4654 and Manni, M., Berkeley, M. R., Seppey, M., & Zdobnov, E. M. (2021). BUSCO: Assessing genomic data quality and beyond.
```
apptainer pull BUSCO.sif docker://quay.io/biocontainers/busco:5.6.1--pyhdfd78af_0
# Checking if BUSCO works
apptainer exec BUSCO.sif busco
```
