# Getting genome file ready for gene prediction

Computing resources and genome
```
exit
sinteractive -c 28 -t 00:15:00 -J Preparation -A PAS2658
cd /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/
cp /fs/scratch/PAS2658/Alexis/Assembly/Polishing/Polishing_Illumina+Nanopore.fasta pf_assembly.fasta
ls -l #make sure it actually copied the file
```

Cleaning assembly - inputting the polished file into Funannotate to clean up the assembly
--exhaustive indicates the level of cleaning to occur
```
apptainer exec /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate clean -i pf_assembly.fasta -o pf_assembly_clean.fasta --exhaustive
```
output
minimap2 version=2.26-r1175 path=/venv/bin/minimap2
ERROR: NODE_386_length_76_cov_2.190476_pilon contains only 3 nucleotides, removing
ERROR: NODE_392_length_62_cov_183.142857_pilon contains only 3 nucleotides, removing
ERROR: NODE_393_length_62_cov_26.142857_pilon contains only 3 nucleotides, removing
ERROR: NODE_395_length_61_cov_11.833333_pilon contains only 3 nucleotides, removing
ERROR: NODE_397_length_60_cov_2.600000_pilon contains only 3 nucleotides, removing
ERROR: NODE_398_length_58_cov_3716.666667_pilon contains only 3 nucleotides, removing
ERROR: NODE_399_length_57_cov_3027.500000_pilon contains only 3 nucleotides, removing
ERROR: NODE_400_length_57_cov_271.500000_pilon contains only 3 nucleotides, removing
ERROR: NODE_401_length_57_cov_49.500000_pilon contains only 2 nucleotides, removing
ERROR: NODE_402_length_56_cov_3703.000000_pilon contains only 3 nucleotides, removing
ERROR: NODE_403_length_56_cov_3149.000000_pilon contains only 1 nucleotides, removing
ERROR: NODE_404_length_56_cov_2718.000000_pilon contains only 3 nucleotides, removing
ERROR: NODE_405_length_56_cov_2425.000000_pilon contains only 3 nucleotides, removing
ERROR: NODE_406_length_56_cov_1094.000000_pilon contains only 1 nucleotides, removing
-----------------------------------------------
408 input contigs, 67 larger than 500 bp, N50 is 3,150,221 bp
Checking duplication of 67 contigs
-----------------------------------------------
-----------------------------------------------
408 input contigs; 67 larger than 500 bp; 0 duplicated; 67 written to file

Renaming the contigs
```

cat pf_assembly_clean.fasta | perl -ane 'if(/\>/){$a++; print ">$a\n"} else {print;}' > pf_assembly_clean_renamed.fasta

```


Sorting and renaming headers
```
apptainer exec /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate sort -i pf_assembly_clean_renamed.fasta -o pf_assembly_clean_renamed_sorted.fasta -m 1000 -b Chr
```
output
67 contigs records loaded
Sorting and renaming contig headers
Removing contigs less than 1000 bp
26 contigs saved to file

Masking the genome: using tantan
```
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate mask --cpus 28 -i pf_assembly_clean_renamed_sorted.fasta -m tantan -o pf_assembly_clean_sorted_masked.fasta
ls
```
output
[Mar 08 10:02 AM]: OS: Debian GNU/Linux 10, 28 cores, ~ 132 GB RAM. Python: 3.8.12
[Mar 08 10:02 AM]: Running funanotate v1.8.17
[Mar 08 10:02 AM]: Soft-masking simple repeats with tantan
[Mar 08 10:02 AM]: Repeat soft-masking finished: 
Masked genome: /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta
num scaffolds: 26
assembly size: 18,972,100 bp
masked repeats: 334,515 bp (1.76%)
