# Training for gene prediction and functional annotation

```
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline
```

# Batch script for Funannotate training # Trainin uses assembled genome to find the structure of genes that are well conserved (orthologs) to use that infor as prediciton model later.
enter the following into the TERMINAL 
timed out, make it 4 hours
```
tee -a Training.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Funannotate_Training
#SBATCH --time=04:00:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline
# Running Funannotate
singularity exec /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate train \
--cpus 28 -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola \
--left \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115198_1_restored_clean_renamed.fastq.gz \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115199_1_restored_clean_renamed.fastq.gz \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115200_1_restored_clean_renamed.fastq.gz \
--right \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115198_2_restored_clean_renamed.fastq.gz \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115199_2_restored_clean_renamed.fastq.gz \
/fs/scratch/PAS2658/Alexis/Annotation/Preprocessing/trimming/SRR8115200_2_restored_clean_renamed.fastq.gz \
--stranded RF --jaccard_clip --species "Peltaster fructicola" --strain LNHT1506
EOF
```
Then, we send it to the scheduler:
```
sbatch Training.sh
squeue -u wafer2
```
this worked 3/8 5pm

Download databases for prediction (you can do this while the training takes place) - takes some time
```
mkdir -p data/database #this is in Funannotate_Pipeline directory
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate setup -i all -w -b all --database data/database
# adding repeats
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate setup -i repeats --database data/database -w -f --database data/database
# adding BUSCO database
cd data/database
wget https://busco-data.ezlab.org/v5/data/lineages/dothideomycetes_odb10.2024-01-08.tar.gz
tar -pxvzf dothideomycetes_odb10.2024-01-08.tar.gz
# Updating databases, just in case
cd ../..
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate setup --update --database data/database
```
I think the training worked...
# Batch script for Funannotate prediction (wait for the training batch job ends and you have all the databses installed)
#The prediction will give the first draft of the structure of the gene.

```
sinteractive -c 28 -t 01:00:00 -J Funnantotate_Pipeline -A PAS2658
# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline

# Making databases available for funannotate
export APPTAINER_BIND="/fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline/data/database:/data/database"

# loading perl
module load perl/5.26.3

# enabling perl modules
module load cpanminus/1.7044

# installing needed module
perl -I $CPANMINUS_INC -Mlocal::lib

cpanm FindBin::Bin
```
output 
! Finding FindBin::Bin on cpanmetadb failed.
! Finding FindBin::Bin () on mirror http://www.cpan.org failed.
! Couldn't find module or a distribution FindBin::Bin
going to keep going to just see
````

# indicating to Funannotate what perl environment to use
perl /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4/change_path_in_perl_scripts.pl "/usr/bin/env perl"
#didnt work, said file not found needed to run (but the file is in there)

# Running Funannotate

apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate predict -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4
```
````
did not work, output
Can't locate FindBin.pm in @INC (you may need to install the FindBin module) (@INC contains: /usr/local/perl/5.26.3/lib/perl5 /users/PAS19
18/wafer2/perl5/lib/perl5/x86_64-linux-thread-multi /users/PAS1918/wafer2/perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.28.1 /usr/local/sh
are/perl/5.28.1 /usr/lib/x86_64-linux-gnu/perl5/5.28 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.28 /usr/share/perl/5.28 /usr/local/lib/site_perl /usr/
lib/x86_64-linux-gnu/perl-base) at /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4/gmes_petap.pl line 79.
BEGIN failed--compilation aborted at /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4/gmes_petap.pl line 79.





Trying something else
```
# Copy container from Jonathanâ€™s working directory:
cp /fs/scratch/PAS2658/Jonathan/Annotation/Funannotate_pipeline/NewFunannotate.sif .

# Then in an interactive session, check if the following command works until the part of GeneMark starts to run:
sinteractive -c 28 -t 01:00:00 -J Funnantotate_Pipeline -A PAS2658

apptainer run NewFunannotate.sif funannotate predict -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /root/gmes_linux_64_4
```

ok now run a diff batch code called Prediction1
````
tee -a Prediction1.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name New_Funannotate_Prediction
#SBATCH --time=03:30:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline

# Making databases available for funannotate
export APPTAINER_BIND="/fs/scratch/PAS2658/Jonathan/Annotation/Funannotate_pipeline/data/database:/data/database"

# loading perl
module load perl/5.26.3

# enabling perl modules
module load cpanminus

# installing needed module
perl -I $CPANMINUS_INC -Mlocal::lib

# indicating to Funannotate what perl environment to use
perl /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4/change_path_in_perl_scripts.pl "/usr/bin/env perl"

# Running New Funannotate
apptainer exec /fs/scratch/PAS2658/Alexis/Annotation/Software/NewFunnanotate.sif funannotate predict -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4
EOF
```
````
now put this into terminal to schedule job
```
sbatch Prediction1.sh
squeue -u wafer2
```
scheduled and executed 11:30 am 3/18/2024
instantly failed
```
tee -a Prediction.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Funannotate_Prediction
#SBATCH --time=01:30:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline
# Making databases available for funannotate
export APPTAINER_BIND="/fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline/data/database/"
module load perl/5.26.3
#perl /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4/change_path_in_perl_scripts.pl "usr/bin/env perl"
# Running Funannotate
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate predict -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /fs/scratch/PAS2658/Alexis/Annotation/Software/gmes_linux_64_4
EOF
```
Then, we send it to the scheduler:
```
sbatch Prediction.sh
squeue -u wafer2
```

one last effort... 

```
tee -a Prediction.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Funannotate_Prediction
#SBATCH --time=01:30:00
#SBATCH --ntasks=28
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline
# Making databases available for funannotate
export APPTAINER_BIND="/fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline/data/database:/data/database"
# Running Funannotate
apptainer exec /fs/scratch/PAS2658/Alexis/Annotation/Software/NewFunannotate.sif funannotate predict -i /fs/scratch/PAS2658/Alexis/Annotation/Genome_Preparation/pf_assembly_clean_sorted_masked.fasta -o Peltaster_fructicola -s "Peltaster fructicola" --strain LNHT1506 --cpus 28 --GENEMARK_PATH /root/gmes_linux_64_4
EOF
```
now send to scheduler

```
sbatch Prediction.sh
squeue -u wafer2
```




# Identify UTRs and update
```
tee -a Update.sh <<EOF
#!/bin/bash
#SBATCH -J ondemand/sys/myjobs/basic_sequential
#SBATCH --job-name Update
#SBATCH --time=03:00:00
#SBATCH --ntasks=2
#SBATCH --exclusive
#SBATCH --mail-type=BEGIN,END,FAIL
#SBATCH --account=PAS2658

# Working directory
cd /fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline
# Making databases available for funannotate
export APPTAINER_BIND="/fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline/data/database:/data/database"
module load perl/5.26.1
# Running Funannotate
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate update -i Peltaster_fructicola --cpus 2
EOF
```
Then, we send it to the scheduler:
```
sbatch Update.sh
squeue -u wafer2
```
# Add the data prediction generated to the database
```
mkdir data/database/trained_species
export APPTAINER_BIND="//fs/scratch/PAS2658/Alexis/Annotation/Funannotate_pipeline/data/database/trained_species:/data/database/trained_species"
apptainer run /fs/scratch/PAS2658/Alexis/Annotation/Software/Funannotate.sif funannotate species -s peltaster_fructicola_lnht1506 -a Peltaster_fructicola/predict_results/peltaster_fructicola_lnht1506.parameters.json
```
